draft-ietf-oauth-selective-disclosure-jwt.md

```diff
@@ -582,8 +602,9 @@ To this end, Verifiers MUST follow the following steps (or equivalent):
        2. Determine the public key for the Holder from the SD-JWT.
        3. Ensure that a signing algorithm was used that was deemed secure for the application. Refer to [@RFC8725], Sections 3.1 and 3.2 for details. The `none` algorithm MUST NOT be accepted.
        4. Validate the signature over the Holder Binding JWT.
-       5. Check that the Holder Binding JWT is valid using `nbf`, `iat`, and `exp` claims, if provided in the Holder Binding JWT.
-       6. Determine that the Holder Binding JWT is bound to the current transaction and was created for this Verifier (replay protection). This is usually achieved by a `nonce` and `aud` field within the Holder Binding JWT.
+       5. Check that the `typ` of the Holder Binding JWT is `hb+jwt`.
+       6. Check that the Holder Binding JWT is valid using `iat`, and, if provided, `nbf` and `exp` claims.
```

Comment:
If `nbf` and `exp` are not provided, then I think this step is not needed. How would somebody check whether the HB-JWT is valid if only `iat` is provided`

Reply:
I think the idea is that the Verifier wants HB-JWTs that are relatively freshly created. And so would reject HB-JWTs that have an `iat` that falls outside some policy based window.

Reply:
it's the same logic as [DPoP JWT](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-dpop-16#section-4.2). iat is sufficient to know if that HB-JWT is fresh enough.

Reply:
I think the idea is that the Verifier wants HB-JWTs that are relatively freshly created. And so would reject HB-JWTs that have an `iat` that falls outside some policy based window.  But something like that should probably said more explicitly because `iat` doesn't have any normative validity rules on its own. Maybe, "Check that Holder Binding JWT was created within an acceptable timeframe as indicated by the `iat` claim" or something along those lines.

FWIW the idea is similar to DPoP and [Checking DPoP Proofs](https://www.ietf.org/archive/id/draft-ietf-oauth-dpop-16.html#name-checking-dpop-proofs) does have some text that attempts to say the the creation time of the JWT needs to be within an acceptable window.

And rather than mentioning the`nbf` and `exp` claims maybe have another general item like "Ensure that the Holder Binding JWT is valid in all other respects per [@!RFC7519]."

Reply:
ok, got it

Reply:
```suggestion
       6. Check that the creation time of the Holder Binding JWT, as determined by the `iat` claim, is within an acceptable window.
       7. Check that the Holder Binding JWT is valid in all other respects, per [@!RFC7519] and [@!RFC8725].
```
updated the verfication steps as suggested.
@awoie @bc-pi

Reply:
wfm